File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/filterCodes/IFilterCodes.java
package org.ngbp.jsonrpc4jtestharness.rpc.filterCodes;

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod;
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType;

import org.ngbp.jsonrpc4jtestharness.rpc.filterCodes.model.GetFilterCodes;

@JsonRpcType
public interface IFilterCodes {

    @JsonRpcMethod("org.atsc.getFilterCodes")
    GetFilterCodes getFilterCodes();

    @JsonRpcMethod("org.atsc.setFilterCodes")
    Object setFilterCodes();
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/filterCodes/IFilterCodes.kt
package org.ngbp.jsonrpc4jtestharness.rpc.filterCodes

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType
import org.ngbp.jsonrpc4jtestharness.rpc.filterCodes.model.GetFilterCodes

@JsonRpcType
interface IFilterCodes {
    @JsonRpcMethod("org.atsc.getFilterCodes")
    fun getFilterCodes(): GetFilterCodes?

    @JsonRpcMethod("org.atsc.setFilterCodes")
    fun setFilterCodes(): Any?
}